File: src/main/java/com/gioia/radiogio/data/repositories/ConfigurationRepositoryImpl.java
package com.gioia.radiogio.data.repositories;

import com.gioia.radiogio.data.domains.Configuration;
import org.dizitart.no2.Nitrite;
import org.dizitart.no2.objects.ObjectRepository;
import org.dizitart.no2.objects.filters.ObjectFilters;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Repository;

@Repository
@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
public class ConfigurationRepositoryImpl implements ConfigurationRepository{
    private Nitrite database;
    private Logger logger = LoggerFactory.getLogger(ConfigurationRepositoryImpl.class);

    private ObjectRepository<Configuration> getRepository(){
        return database.getRepository(Configuration.class);
    }

    @Override
    public void upsert(Configuration configuration) {
        getRepository().update(configuration, true);
        logger.atDebug().log("Actualizada la configuración ${configuration.key}, valor: ${configuration.value}");
    }

    @Override
    public Configuration find(String key) {
        logger.atDebug().log("Buscando configuración $key");
        Configuration config = getRepository()
            .find(
                ObjectFilters.eq("key", key)
            )
            .firstOrDefault();
        logger.atDebug().log("Leída la configuración $key, valor: " + (config.getValue() != null ? config.getValue() : "n.d"));
        return config;
    }
}

File: src/main/kotlin/com/gioia/radio/data/repositories/ConfigurationRepositoryImpl.kt
package com.gioia.radio.data.repositories

import com.gioia.radio.data.domains.Configuration
import org.dizitart.no2.IndexOptions
import org.dizitart.no2.IndexType
import org.dizitart.no2.Nitrite
import org.dizitart.no2.objects.filters.ObjectFilters
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class ConfigurationRepositoryImpl(
    private val database: Nitrite,
) : ConfigurationRepository {
    private var logger: Logger = LoggerFactory.getLogger(ConfigurationRepositoryImpl::class.java)

    override fun upsert(configuration: Configuration) {
        database
            .getRepository(Configuration::class.java)
            .update(configuration, true)
        logger.atDebug().log("Actualizada la configuración ${configuration.key}, valor: ${configuration.value}")
    }

    override fun find(key: String): Configuration? {
        logger.atDebug().log("Buscando configuración $key")
        val config = database
            .getRepository(Configuration::class.java)
            ?.find(
                ObjectFilters.eq("key", key)
            )
            ?.firstOrDefault()
        logger.atDebug().log("Leída la configuración $key, valor: ${config?.value?: "n.d"}")
        return config
    }

    override fun createIndexes(){
        val repository = database.getRepository(Configuration::class.java)
        if (repository != null && !repository.hasIndex("key")) {
            repository.createIndex("key",
                IndexOptions.indexOptions(IndexType.Unique, false)
            )
        }
    }
}